src/Cake.Core.Tests/Unit/IO/NuGet/NuGetAssemblyCompatibilityFilterTests.cs

```diff
@@ -0,0 +1,126 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+using Cake.Core.IO;
+using Cake.Core.IO.NuGet;
+using NSubstitute;
+using Xunit;
+
+namespace Cake.Core.Tests.Unit.IO.NuGet
+{
+    public class NuGetAssemblyCompatibilityFilterTests
+    {
+        public sealed class TheFilterCompatibleAssembliesMethod
+        {
+            private static readonly FrameworkName _dummyFrameworkName = new FrameworkName(".NETFramework,Version=v4.5.1");
+
+            [Fact]
+            public void Should_Throw_If_Target_Framework_Is_Null()
+            {
+                // Given
+
+                var filter = new NuGetAssemblyCompatibilityFilter(Substitute.For<INuGetFrameworkCompatibilityFilter>(),
+                    Substitute.For<IPackageReferenceBundler>());
+
+                // When
+                // ReSharper disable once ExpressionIsAlwaysNull
+                var result = Record.Exception(() => filter.FilterCompatibleAssemblies(null, new FilePath[0]));
+
+                // Then
+                Assert.IsArgumentNullException(result, "targetFramework");
+            }
+
+            [Fact]
+            public void Should_Throw_If_Assembly_Paths_Is_Null()
+            {
+                // Given
+                var filter = new NuGetAssemblyCompatibilityFilter(Substitute.For<INuGetFrameworkCompatibilityFilter>(),
+                    Substitute.For<IPackageReferenceBundler>());
+
+                // When
+                var result = Record.Exception(() => filter.FilterCompatibleAssemblies(_dummyFrameworkName, null));
+
+                // Then
+                Assert.IsArgumentNullException(result, "assemblyPaths");
+            }
+
+            [Fact]
+            public void Should_Return_Empty_When_No_References_Are_Compatible()
+            {
+                // Given
+                var targetFramework = _dummyFrameworkName;
+
+                var compatibilityFilter = Substitute.For<INuGetFrameworkCompatibilityFilter>();
+
+                compatibilityFilter.GetCompatibleItems(targetFramework, Arg.Any<IEnumerable<PackageReferenceSet>>())
+                    .Returns(Enumerable.Empty<PackageReferenceSet>());
+
+                var referenceSetFactory = Substitute.For<IPackageReferenceBundler>();
+
+                var assemblyFiles = new FilePath[] { "dummy.dll", "dummy2.dll" };
+                var filter = new NuGetAssemblyCompatibilityFilter(compatibilityFilter,
+                    referenceSetFactory);
+
+                // When
+                var result = filter.FilterCompatibleAssemblies(targetFramework, assemblyFiles);
+
+                // Then
+                Assert.Empty(result);
+            }
+
+            [Fact]
+            public void Should_Throw_If_Any_Assembly_Path_Is_Not_Relative()
```

Comment:
This test is new, added to document the previously implicit requirement that assembly paths must be relative to the package directory in order to be parsed properly.